File: tutorial/tribot/src/data/Variables.java
package data;

import data.structures.PrivateSingleton;
import script.TutorialIsland;

public class Variables extends PrivateSingleton {

    private boolean pickedGender;
    private String username;

    @Override
    public Class getAllowedClass() {
        return TutorialIsland.class;
    }

    public Variables(Class callingClass) throws IllegalAccessException {
        super(callingClass);
        pickedGender = false;
    }

    public boolean hasPickedGender() {
        return pickedGender;
    }

    public void setPickedGender(boolean pickedGender) {
        this.pickedGender = pickedGender;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }
}

File: scripts/tutorial/src/scripts/tutorial/data/Variables.kt
package scripts.tutorial.data

import scripts.data.structures.PrivateSingleton
import scripts.tutorial.missions.TutorialIsland

class Variables(callingClass: Class<*>?) : PrivateSingleton(callingClass) {
    private var pickedGender = false
    @JvmField
    var username: String? = null
    override fun getAllowedClass(): Class<*> {
        return TutorialIsland::class.java
    }

    fun hasPickedGender(): Boolean {
        return pickedGender
    }

    fun setPickedGender(pickedGender: Boolean) {
        this.pickedGender = pickedGender
    }
}